rosette/h/Prim.h

```diff
@@ -124,6 +124,10 @@ class BuiltinPrim {
         return PRIM_MISMATCH((n), _STRING(type)); \
     type* var = (type*)ARG(n);
 
+#define CHECK_NOVAR(n, type)                      \
+    if (!IS_A(ARG(n), type))                      \
+        return PRIM_MISMATCH((n), _STRING(type)); \
+
```

Comment:
Might not be worth it since you control all invocation sites, but this compiles, is supposed to be standard C++11, and only evaluates the numeric argument once:
```C++
#define CHECK_NOVAR(n, type)                            \
    do {                                                \
        decltype(n) __n = (n);                          \
        if (!IS_A(ARG(__n), type)) {                    \
            return PRIM_MISMATCH((__n), _STRING(type)); \
        }                                               \
    } while (0)
```

Reply:
I like it!